### ISSUE TITLE
Tried on two machines but can't get grab-all-posters.py to run 

### ISSUE THREAD

**Issue by isaacolsen94** [chazlarson/Media-Scripts#38]
Here is the output I got on put an Ubuntu VM and Windows 11 main desktop: 
```
Starting grab-all-posters 0.8.8 at 2024-02-27 19:41:21
Threaded downloads: True
connecting to http://10.10.10.30:32400...
Loading Movies ...
Resetting rundate for Movies to None...
Skipping collection artwork ...
Loading everything in collection Harry Potter ...
Completed loading 10 from collection Harry Potter
on 0: Problem processing David Holmes: The Boy Who Lived; Path.replace() takes 2 positional arguments but 3 were given
on 1: Problem processing Harry Potter and the Sorcerer's Stone; Path.replace() takes 2 positional arguments but 3 were given
on 2: Problem processing Harry Potter and the Chamber of Secrets; Path.replace() takes 2 positional arguments but 3 were given
on 3: Problem processing Harry Potter and the Prisoner of Azkaban; Path.replace() takes 2 positional arguments but 3 were given
on 4: Problem processing Harry Potter and the Goblet of Fire; Path.replace() takes 2 positional arguments but 3 were given
on 5: Problem processing Harry Potter and the Order of the Phoenix; Path.replace() takes 2 positional arguments but 3 were given
on 6: Problem processing Harry Potter and the Half-Blood Prince; Path.replace() takes 2 positional arguments but 3 were given
on 7: Problem processing Harry Potter and the Deathly Hallows: Part 1; Path.replace() takes 2 positional arguments but 3 were given
on 8: Problem processing Harry Potter and the Deathly Hallows: Part 2; Path.replace() takes 2 positional arguments but 3 were given
on 9: Problem processing Harry Potter 20th Anniversary: Return to Hogwarts; Path.replace() takes 2 positional arguments but 3 were given
Grab all posters Movies |████████████████████████████████████████| 10/10 [100%] in 0.1s (138.33/s)
Processed 0 of 10
Loading TV Shows ...
Resetting rundate for TV Shows to None...
Skipping collection artwork ...
Loading everything in collection Harry Potter ...
Completed loading 0 from collection Harry Potter
Problem processing TV Shows; name 'last_run_season' is not defined
Complete!
```
Here is my .env file:
```
# PLEX API ENV VARS
PLEXAPI_PLEXAPI_TIMEOUT='360'
PLEXAPI_AUTH_SERVER_BASEURL=http://10.10.10.30:32400
                                             # Just the base URL, no /web or anything at the end.
                                             # i.e. http://192.168.1.11:32400 or the like
PLEXAPI_AUTH_SERVER_TOKEN=*******************
PLEXAPI_LOG_BACKUP_COUNT='3'
PLEXAPI_LOG_FORMAT='%(asctime)s %(module)12s:%(lineno)-4s %(levelname)-9s %(message)s' # PLEX API ENV VARS
PLEXAPI_LOG_LEVEL='INFO'
PLEXAPI_LOG_PATH='plexapi.log'
PLEXAPI_LOG_ROTATE_BYTES='512000'
PLEXAPI_LOG_SHOW_SECRETS='false'

# GENERAL ENV VARS
TMDB_KEY=*******************    # https://developers.themoviedb.org/3/getting-started/introduction
TVDB_KEY=TVDB_V4_API_KEY                     # currently not used; https://thetvdb.com/api-information
DELAY=1                                      # optional delay between items
LIBRARY_NAMES='Movies,TV Shows'                # comma-separated list of libraries to act on

# IMAGE DOWNLOAD ENV VARS
## what-to-grab
GRAB_SEASONS=1                               # should get-all-posters retrieve season posters?
GRAB_EPISODES=1                              # should get-all-posters retrieve episode posters? [requires GRAB_SEASONS]
GRAB_BACKGROUNDS=1                           # should get-all-posters retrieve backgrounds?
ONLY_CURRENT=0                               # should get-all-posters retrieve ONLY current artwork?
ARTWORK=1                                    # current background is downloaded with current poster
INCLUDE_COLLECTION_ARTWORK=0                 # should get-all-posters retrieve collection posters?
ONLY_COLLECTION_ARTWORK=0                    # should get-all-posters retrieve ONLY collection posters?
ONLY_THESE_COLLECTIONS='Harry Potter'                      # only grab artwork for these collections and items in them
POSTER_DEPTH=0                               # grab this many posters [0 grabs all]
KEEP_JUNK=0                                  # keep files that script would normally delete [incorrect filetypes, mainly]
FIND_OVERLAID_IMAGES=0                       # check all downloaded images for overlays
# RETAIN_OVERLAID_IMAGES=0                   # keep images that have an overlay EXIF tag [this will override the following two]
RETAIN_PMM_OVERLAID_IMAGES=0                 # keep images that have the PMM overlay EXIF tag 
RETAIN_TCM_OVERLAID_IMAGES=0                 # keep images that have the TCM overlay EXIF tag 

## where-to-put-it
USE_ASSET_NAMING=0                           # should grab-all-posters name images to match PMM's Asset Directory requirements?
USE_ASSET_FOLDERS=0                          # should those PMM-Asset-Directory names use asset folders?
USE_ASSET_SUBFOLDERS=0                       # create asset folders in subfolders ["Collections", "Other", or [0-9, A-Z]] ]
ASSETS_BY_LIBRARIES=1                        # should those PMM-Asset-Directory images be sorted into library folders?
ASSET_DIR=D:\Temp\plex_poster\assets                             # top-level directory for those PMM-Asset-Directory images
                                             # if asset-directory naming is on, the next three are ignored                                            
POSTER_DIR=D:\Temp\plex_poster\download            # put downloaded posters here
CURRENT_POSTER_DIR=D:\Temp\plex_poster\current     # put downloaded current posters and artwork here
POSTER_CONSOLIDATE=0                         # if false, posters are separated into folders by library

## tracking
TRACK_URLS=1                                 # If set to 1, URLS are tracked and won't be downloaded twice
TRACK_COMPLETION=1                           # If set to 1, movies/shows are tracked as complete by rating id
TRACK_IMAGE_SOURCES=1                        # keep a file containing file names and source URLs

## general
POSTER_DOWNLOAD=1                            # if false, generate a script rather than downloading
FOLDERS_ONLY=1                               # Just build out the folder hierarchy; no image downloading
DEFAULT_YEARS_BACK=0                         # in absence of a "last run date", grab things added this many years back.
                                             # 0 sets the fallback date to the beginning of time
THREADED_DOWNLOADS=1                         # should downloads be done in the background in threads?
RESET_LIBRARIES=Movies,TV Shows              # reset "last time" count to the fallback date for these libraries
RESET_COLLECTIONS=                           # CURRENTLY UNUSED
ADD_SOURCE_EXIF_COMMENT=1                    # CURRENTLY UNUSED
```

I'm lost of figuring out what is going wrong

**Comment by chazlarson** [comment]
Fixed a logic error; try 0.8.9

**Comment by chazlarson** [comment]
issue fixed according to discord
